lib/elixir/lib/enum.ex

```diff
@@ -899,14 +899,18 @@ defmodule Enum do
   end
 
   def find_index(enumerable, fun) when is_function(fun, 1) do
-    res =
-      Enumerable.reduce(enumerable, {:cont, 0}, fn(entry, acc) ->
-        if fun.(entry), do: {:halt, acc}, else: {:cont, acc + 1}
-      end)
+    res = Enumerable.reduce(enumerable, {:cont, {0, 0}}, fn(entry, acc) ->
```

Comment:
Also, let's not change the style here: let's leave the `Enumerable.reduce/3` call on the next line :)